app/commands/solution/complete.rb

```diff
@@ -37,17 +37,14 @@ def record_activity!
       Rails.logger.error e.message
     end
 
-    def guard!
-      raise SolutionNotFoundError unless solution
-      raise UserTrackNotFoundError unless user_track
-    end
```

Comment:
The fact that these are now gone is a strong indicator that is a good refactoring! :)

Reply:
Yes! Good point :D